plumbing/format/packfile/decoder.go

```diff
@@ -53,7 +53,15 @@ type Decoder struct {
 	crcs         map[plumbing.Hash]uint32
 }
 
-// NewDecoder returns a new Decoder that reads from r.
+// NewDecoder returns a new Decoder that reads from s and store the objects in
+// o. ObjectStorer can be nil, in this case the objects are not stored but
+// Offsets can be call to retrieve the objets offset in the packfile being scan.
+//
+// If ObjectStorer is nil and the Scanner is not Seekable, ErrNonSeekable is
+// returned.
+//
+// If the ObjectStorer implements storer.Transactioner, a transation is created
+// during the Decode execution, if something fails the Rollback is called
```

Comment:
It will be nice to explain all the possible combinations of `ObjectStorer`s and seekable scanners, from the very beginning, just to make it easier later to document the internal parts.